devito/ir/clusters/algorithms.py

```diff
@@ -283,26 +283,41 @@ def callback(self, clusters, prefix, backlog=None, known_break=None):
         return processed + self.callback(backlog, prefix)
 
 
-def optimize(clusters):
+def optimize(clusters, dse_mode):
     """
     Optimize a topologically-ordered sequence of Clusters by applying the
     following transformations:
 
-        * Fusion
-        * Lifting
-
-    Notes
-    -----
-    This function relies on advanced data dependency analysis tools based upon classic
-    Lamport theory.
+        * [cross-cluster] Fusion
+        * [intra-cluster] Several flop-reduction passes via the DSE
+        * [cross-cluster] Lifting
+        * [cross-cluster] Scalarization
+        * [cross-cluster] Arrays Elimination
     """
+    # To create temporaries
+    counter = generator()
+    template = lambda: "r%d" % counter()
```

Comment:
Stupid question but why `r`?

Reply:
not sure... stupid choice from three years ago ?